v2/index/indexmhsorted.go

```diff
@@ -0,0 +1,210 @@
+package index
+
+import (
+	"encoding/binary"
+	"io"
+	"sort"
+
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multicodec"
+	"github.com/multiformats/go-multihash"
+)
+
+// TODO replace with propper multicodec once defined.
+const IndexMultihashSortedCodec = multicodec.Code(0x0401)
+
+type (
+	multiWidthCodedIndex  map[uint32]codedSingleWidthIndex
+	codedSingleWidthIndex map[uint64]singleWidthIndex
```

Comment:
do we need two layers (bitwidth and then multihash code)? can we get away with only splitting the index by the multihash algorithm, and assume the width within a given multihash algorithm will always be the same?

Reply:
Agreed; marking the PR as draft to get this change through. Thanks Will